substratevm/src/com.oracle.svm.agent/src/com/oracle/svm/agent/BreakpointInterceptor.java

```diff
@@ -1043,67 +1053,78 @@ private static JNIObjectHandle shouldIncludeMethod(JNIEnvironment jni, JNIObject
         if (clearException(jni)) {
-            validObjectStreamClassInstance = false;
+            validCapturingClass = false;
         }
 
-        // Skip Lambda class serialization
-        if (serializeTargetClassName.contains("$$Lambda$")) {
-            return true;
-        }
+        traceSerializeBreakpoint(jni, "SerializedLambda.readResolve", validCapturingClass, state.getFullStackTraceOrNull(), capturingClassName);
+        return true;
+    }
 
-        List<String> transitiveSerializeTargets = new ArrayList<>();
-        transitiveSerializeTargets.add(serializeTargetClassName);
+    private static boolean objectStreamClassConstructor(JNIEnvironment jni, Breakpoint bp, InterceptedState state) {
 
-        /*
-         * When the ObjectStreamClass instance is created for the given serializeTargetClass, some
-         * additional ObjectStreamClass instances (usually the super classes) are created
-         * recursively. Call ObjectStreamClass.getClassDataLayout0() can get all of them.
-         */
-        JNIMethodId getClassDataLayout0MId = agent.handles().getJavaIoObjectStreamClassGetClassDataLayout0(jni, bp.clazz);
-        JNIObjectHandle dataLayoutArray = Support.callObjectMethod(jni, objectStreamClassInstance, getClassDataLayout0MId);
-        if (!clearException(jni) && nullHandle().notEqual(dataLayoutArray)) {
-            int length = jniFunctions().getGetArrayLength().invoke(jni, dataLayoutArray);
-            // If only 1 element is got from getClassDataLayout0(). it is base ObjectStreamClass
-            // instance itself.
-            if (!clearException(jni) && length > 1) {
-                JNIFieldId hasDataFId = agent.handles().getJavaIOObjectStreamClassClassDataSlotHasData(jni);
-                JNIFieldId descFId = agent.handles().getJavaIOObjectStreamClassClassDataSlotDesc(jni);
-                JNIMethodId javaIoObjectStreamClassForClassMId = agent.handles().getJavaIoObjectStreamClassForClass(jni, bp.clazz);
-                for (int i = 0; i < length; i++) {
-                    JNIObjectHandle classDataSlot = jniFunctions().getGetObjectArrayElement().invoke(jni, dataLayoutArray, i);
-                    boolean hasData = jniFunctions().getGetBooleanField().invoke(jni, classDataSlot, hasDataFId);
-                    if (hasData) {
-                        JNIObjectHandle oscInstanceInSlot = jniFunctions().getGetObjectField().invoke(jni, classDataSlot, descFId);
-                        if (!jniFunctions().getIsSameObject().invoke(jni, oscInstanceInSlot, objectStreamClassInstance)) {
-                            JNIObjectHandle oscClazz = Support.callObjectMethod(jni, oscInstanceInSlot, javaIoObjectStreamClassForClassMId);
-                            String oscClassName = getClassNameOrNull(jni, oscClazz);
-                            transitiveSerializeTargets.add(oscClassName);
+        JNIObjectHandle serializeTargetClass = getObjectArgument(1);
+        if (isSerializable(jni, serializeTargetClass)) {
```

Comment:
When serializing lambda class, it's capturing class is also being registered for the serialization since the method com.oracle.svm.agent.BreakpointInterceptor#objectStreamClassConstructor is being called for that class too. In some cases, that class might not be serializable. We simply don't want to add it to the configuration file. Everything would work, but we want to avoid warning in the com.oracle.svm.reflect.serialize.hosted.SerializationFeature$SerializationBuilder#registerWithTargetConstructorClass.